### ISSUE TITLE
Training strategy for AI-aid unwrapping package

### ISSUE THREAD

**Issue by rogerkuou** [MotionbyLearing/Planning#4]

**Comment by rogerkuou** [comment]
A first implementation of the Recurrent Neuron Network method can be found in the [paper on Slack](https://mobyleworkspace.slack.com/files/U03A6EHH3DZ/F03BD74L0F6/tudelft_rnn_manuscript_for_mobyle_team.pdf).

In this paper, an RNN with 3 LSTM layers is trained to predict deformation from time (DOY), precipitation, and temperature. These three variables are used as input, while the extensometer data, which is regarded as the true value, is used as the output (target value).

This RNN model is then benchmarked/tested against another extensometer data to derive the emission probability, which is one component to make the decision to add/minus a cycle or not.

In this way, RNN aids the unwrapping algorithms by contributing to the emission probability, rather than directly to the results. The emission probability is static per the RNN model per benchmarking data.

Several points to consider in this paper:

The RNN is trained based on the "true value" (extensometer data). How can we tell if the improvement of unwrapping is contributed by contextual information or the true value?
The RNN takes into account the precipitation and temperature. Are they considered as the only two major driven mechanisms of the deformation? Can we also use the wrapped phase as one of the inputs?

**Comment by rogerkuou** [comment]
Based on the discussion with @FreekvanLeijen today, we think there are three key points that need to be considered in the training strategy:

1. The integer nature of the phase shift. We would like the model to predict the add/subtract the integer number of phase cycles, rather than predict the continuous phase value.
2. The way of handling contextual information. Different types of contextual information need to be considered for different types of scatters. This means we may need to cluster the scatters before training the regression model, and consider different training strategies for each cluster.
3. We would like the predict the deformation with uncertainty per epoch. There are two potential ways to do this: 1) train a conventional model and do a sensitivity analysis afterward, or use the Bayesian Neuron Network (BNN) to learn the behavior of uncertainty. The former way requires the model to be linearizable, the latter way requires a large amount of training data to make the model converge. We can test both ways and decide later.

@sonjageorgievska Maybe we can discuss about this before our meeting next week?

**Comment by rogerkuou** [comment]
Freek has intially defined three sections of using ML to improve the InSAR processing. Below are the points we can follow to discuss per section:

1. Point selection / clustering
In this step, we would like to classify wrapped InSAR scatters into groups.
- What is the essential definition of a cluster? Is it a certain surface type? Or a group of scatters following similar deformation pattern?
    - If it’s the former one, we may not even need to involve SAR data in this stage. Since the surface information can be extracted straight forwardly from the contextual data.
    - If it’s the latter one, we do need to involve the SAR data
- Do we expect the SNR level of raw phase to be high enough to reach a meaningful cluster result?
- Do we want to apply the clustering to all the raw pixels, e.g. all scatters not on water body? If this is a clustering based on SAR data, this volume may be too large.

2. Time series unwrapping
In this step we would like to train a model to predict the phase unwrapping decisions.
- What is the label data (the ground truth?) We may have several choices:
    - Expert interpreted data. (Model will be biased toward the manual interpretation. But may not be a problem.)
    - Simulated data. (Model will be biased to the simulated data)
    - Higher precision observations, e.g. GPS, extensometer. (The volume is very limited )- 
- What is the application extent of a model? Do we train a model for all the scatters? Or a model per group?
    - If there is a model per group, how do we define this group?
    - Does this group have to be the same as the previous step?

3. Time series interpretation
- How does this interpretation influence the pre-clustering results?
    - Shall the algorithm be an iterative process?
    - If it is for interpretation, is it fair to still involve the contextual data? Since it is already used in step 1?
